src/vm/i386/cgenx86.cpp

```diff
@@ -402,6 +402,11 @@ void HelperMethodFrame::UpdateRegDisplay(const PREGDISPLAY pRD)
 
     ClearRegDisplayArgumentAndScratchRegisters(pRD);
 
+    //
+    // Fix up ResumeSp
+    //
+    pRD->pCurrentContext->ResumeEsp = EECodeManager::GetResumeSp(pRD);
```

Comment:
This assumes that the FCall is always called from JITed code. It is not always the case. For example, the FCall can be called by CallDescrWorker.